src/General-Rules-Tests/ReInconsistentMethodClassificationRuleTest.class.st

```diff
@@ -0,0 +1,40 @@
+Class {
+	#name : 'ReInconsistentMethodClassificationRuleTest',
+	#superclass : 'ReAbstractRuleTestCase',
+	#category : 'General-Rules-Tests-Migrated',
+	#package : 'General-Rules-Tests',
+	#tag : 'Migrated'
+}
+
+{ #category : 'tests' }
+ReInconsistentMethodClassificationRuleTest >> testRule [
+
+	| critiques |
+	self class
+		compile: 'myCritiques super myCritiques'
+		classified: 'accessing'.
+	critiques := self myCritiquesOnMethod: self class >> #myCritiques.
+	self assert: critiques size equals: 1.
+	[ "code of the test" ] ensure: [
+		(self class >> #myCritiques) removeFromSystem ].
+	self class
+		compile: 'initialize super initialize'
+		classified: 'accessing'.
+	critiques := self myCritiquesOnMethod: self class >> #initialize.
+	self assert: critiques size equals: 1.
+	[ "code of the test" ] ensure: [
```

Comment:
```suggestion
	| critiques |
	self class
		compile: 'myCritiques super myCritiques'
		classified: 'accessing'.
	[ critiques := self myCritiquesOnMethod: self class >> #myCritiques.
	self assert: critiques size equals: 1 ] ensure: [
		(self class >> #myCritiques) removeFromSystem ].
	self class
		compile: 'initialize super initialize'
		classified: 'accessing'.
	[ critiques := self myCritiquesOnMethod: self class >> #initialize.
	self assert: critiques size equals: 1 ] ensure: [
```

The #ensure: method makes sure the argument blocks gets executed after the block receiving the message, even if the block does not finish (eg. a non local return or an exception happens). So the receiving block must not be empty.